### PR TITLE
feat(hooks): add UI warning for BeforeTool hooks in confirmation dialog

### LINKED ISSUE #15167: Fix execution order of BeforeTool
`BeforeTool` runs after tool confirmation - if a hook is going to block the tool call, we should just block it, not ask the user then block it. 

Update the sequencing of the hook execution

### PR DESCRIPTION
**Pull request by afarber**
## Summary

Add a warning message in the tool confirmation dialog when a BeforeTool hook might block the tool after user confirmation

## Details

When a BeforeTool hook is configured that matches the current tool, the confirmation dialog now displays:

> Note: A hook may modify or block this action after confirmation.

Changes:

- Added `mayBeBlockedByHook` field to all confirmation detail interfaces
- Added hook detection logic in `CoreToolScheduler` before showing confirmation
- Added warning UI in `ToolConfirmationMessage` component

## Related Issues

Fixes #15167

## How to Validate

1. Configure a BeforeTool hook in `.gemini/settings.json`
2. Trigger a tool that requires confirmation (e.g., shell command)
3. Verify the warning message appears in the confirmation dialog

## Pre-Merge Checklist

1. Run tests: `npm run test --workspace @google/gemini-cli-core -- src/core/coreToolScheduler.test.ts -t "BeforeTool"`
2. Run hook type tests: `npm run test --workspace @google/gemini-cli-core -- src/hooks/types.test.ts`
3. Configure a BeforeTool hook that returns `{ "decision": "block" }` and verify the tool is blocked without showing a confirmation dialog

- [ ] Updated relevant documentation and README (if needed)
- [x] Added/updated tests (if needed)
- [ ] Noted breaking changes (if any)
- [x] Validated on required platforms/methods:
  - [ ] MacOS
    - [ ] npm run
    - [ ] npx
    - [ ] Docker
    - [ ] Podman
    - [ ] Seatbelt
  - [ ] Windows
    - [ ] npm run
    - [ ] npx
    - [ ] Docker
  - [x] Linux
    - [x] npm run
    - [ ] npx
    - [ ] Docker

### CODE REVIEW

File: packages/core/src/hooks/types.ts [hunk 1]
```diff
@@ -250,6 +257,24 @@ export class BeforeToolHookOutput extends DefaultHookOutput {
...
+      'permissionDecision' in this.hookSpecificOutput
+    ) {
+      const compatDecision = this.hookSpecificOutput['permissionDecision'];
+      if (compatDecision === 'approve' || compatDecision === 'allow') {
+        return true;
+      }
+    }
+
+    return super.isApprovingDecision();
+  }
```
[the remark this answers is not in the document]
Comment (afarber) [reply packages/core/src/hooks/types.ts]: Fixed! Both isBlockingDecision() and isApprovingDecision() now return based solely on permissionDecision when it exists, without falling through to the base class. Added tests for these edge cases.

File: packages/core/src/core/coreToolScheduler.ts [hunk 2]
```diff
@@ -878,6 +880,55 @@ export class CoreToolScheduler {
...
+              signal,
+              createErrorResponse(
+                reqInfo,
+                new Error(`Agent execution stopped by hook: ${reason}`),
+                ToolErrorType.EXECUTION_FAILED,
+              ),
+            );
+            await this.checkAndNotifyCompletion(signal);
+            return;
+          }
```
[the remark this answers is not in the document]
Comment (afarber) [reply packages/core/src/core/coreToolScheduler.ts]: Good catch! Added getBlockingOrStoppingResult() helper method to DefaultHookOutput and refactored both coreToolScheduler.ts and coreToolHookTriggers.ts to use it. Also added tests for the new helper.
